controllers/bsl.go

```diff
@@ -284,7 +284,7 @@ func (r *DPAReconciler) validateAWSBackupStorageLocation(bslSpec velerov1.Backup
 		return fmt.Errorf("prefix for AWS backupstoragelocation object storage cannot be empty. It is required for backing up images")
 	}
 	// BSL region is required when s3ForcePathStyle is true AND BackupImages is true
-	if (bslSpec.Config == nil || len(bslSpec.Config[Region]) == 0 && bslSpec.Config[S3ForcePathStyle] == "true") && dpa.BackupImages() {
+	if (bslSpec.Config == nil || len(bslSpec.Config[Region]) == 0 || (len(bslSpec.Config[Region]) == 0 && bslSpec.Config[S3ForcePathStyle] == "true")) && dpa.BackupImages() {
```

Comment:
```
len(bslSpec.Config[Region]) == 0
 || 
(len(bslSpec.Config[Region]) == 0
```

wait what does this do?

Reply:
ahh.... nvm I see..
```
(len(bslSpec.Config[Region]) == 0 && bslSpec.Config[S3ForcePathStyle] == "true")
```

Reply:
from the error message, I think this is wrong

example: if I set backup images to false, this if is never triggered, right?

to make sure, can you add one more test with `BackupImages` false but with `S3ForcePathStyle` true and no region (only one is this I think https://github.com/openshift/oadp-operator/blob/master/controllers/bsl_test.go#L1141)

Reply:
yes if `backupImages` is false, we will not get the error.

Reply:
updated validation and added more unit tests

Reply:
can we simplify the second if? something like (merging the 2 together)
```go
       // BSL region is required when s3ForcePathStyle is true or BackupImages is true
	if len(bslSpec.Config[Region]) == 0 {
		if bslSpec.Config[S3ForcePathStyle] == "true" {
			return fmt.Errorf("region for AWS backupstoragelocation cannot be empty when s3ForcePathStyle is true")
		}
		if dpa.BackupImages() {
			return fmt.Errorf("region for AWS backupstoragelocation cannot be empty when BackupImages is true")
		}
	}
```

Reply:
This would not cover the case when region is empty and `S3ForcePathStyle` is `true` alongwith `backupImages` is enabled.

Reply:
The error message would not tell all info, but it would be covered by `s3ForcePathStyle=true` case, right?

Reply:
current PR covers the case